### ISSUE TITLE
Almost no compatible legacy themes on TB >= 1.0.4

### ISSUE THREAD

**Issue by doclucas** [thirtybees/thirtybees#590]
While working on the [TB Mega Compatibility list](https://docs.google.com/spreadsheets/d/1eXznv2HEBEojtr1_JaR0r4m-Z3N5k_vU7eLYr5B_wb0/edit?usp=sharing), it became very clear that there are no more themes compatible with TB, except for Warehouse 3, and of course the default community theme. It seems that TB has become even less compatible with the launch of v1.0.4 when the only 2 additional themes that did work stopped working correctly  - Panda & Transformer.
Backward compatibility, _including with legacy themes_, is critical for the adoption of and migration to Thirtybees, especially for merchants who have already invested in PS 1.6 Themes and have live sites with those themes and of course for any merchant who want to have different themes for their site (eg. _most_ merchants).
I noticed that (thankfully) @getdatakick is on the blitz of theme related issues and compatibility issues fixes and thus it's a perfect time to bring this up, even more so now before PS 1.6 support ends.
I think getting the core to be fully backward compatible with legacy themes should be extremely high on the project priorities and will draw a lot of merchants to shift to TB.

P.S.
DataKick, can't thank you enough. You are doing an amazing job!

**Comment by Traumflug** [comment]
If it's just the `Config.xml` vs. `config.xml` bug I'd still consider these themes compatible. That's something merchants can deal with and as you noted correctly, this bug is also about to go away.

**Comment by Dh42** [comment]
There are a ton of themes that are compatible, but just are not listed compatible too. So that is an issue, we need to apply more pressure to the theme makers.

**Comment by doclucas** [comment]
>If it's just the Config.xml vs. config.xml bug I'd still consider these themes compatible. That's something merchants can deal with and as you noted correctly, this bug is also about to go away.

I wish it was that simple. It's definitely _not_ the `Config.xml` issue on Panda and Transformer themes and not the issue on several other PS 1.6 themes I tried. The new v1.0.4 broke compatibility in other ways. Please don't take my word for it, test them yourself.

>There are a ton of themes that are compatible, but just are not listed compatible too. So that is an issue, we need to apply more pressure to the theme makers.

Which specific themes are compatible with **>=v1.0.4**, Lesley? If you know any that are proven to work perfectly, do tell. This information is absolutely essential, for the compatibility list at the very least. 
Unfortunately, I personally found none that is 100% compatible, other than Warehouse 3 that got confirmed reports by forum members and I tested and confirmed it myself as well.

**Comment by Traumflug** [comment]
> The new v1.0.4 broke compatibility in other ways.

That's a bug, then, and reports are welcome. Duplicating testing efforts is a bit, well, inefficient.

And, uhm, "the new" is *1.0.6*, not *1.0.4*.

**Comment by Dh42** [comment]
Every template monster 1.6 theme should be compatible, we both use the same core template. What issues are you having with compatibility with panda / transformer. That might be the best place to start.

**Comment by doclucas** [comment]
Kindly read my edited post.
Markus, I know the latest is 1.0.6, I assume it's quite clear I meant that the issues stared since 1.0.4

**Comment by Dh42** [comment]
Can you explain the issue that you are seeing. What error are you getting, things that would help lead us in the direction of solving the problem.

**Comment by doclucas** [comment]
Did you read the links?
Quote:
>The current version of Panda installed on TB 1.0.4 causes an incorrect display of the modal window with the confirmation of adding to the basket (eg no product name).
On TB 1.0.3 this error does not occur.

**Comment by doclucas** [comment]
If the same exact theme works perfectly fine on 1.0.3 and not on 1.0.4, how is that not a bug on our end?

**Comment by doclucas** [comment]
Moreover, when prospects read something like this:
https://lh3.googleusercontent.com/-49iFuE5jQiA/W5FJURn-YLI/AAAAAAAAKxw/jSr4WwMieCoD6q7EHDed02Y1xjhLYLbZwCHMYCw/s0/chrome_2018-09-06_22-35-44.png

It, by itself, clearly suggests that TB is not compatible with PS 1.6 out of the box and theme makers need to make their themes compatible (which is not so likely to happen in most cases without sufficient demand), or we have to make our end more compatible to bring more prospects, which is the reason for opening this issue.

**Comment by Dh42** [comment]
Why not reach out to him and ask him to change it? Have you yet?

**Comment by doclucas** [comment]
I think reaching out to TB developers to make the core more compatible is much more efficient that reaching out to every theme maker in the world. Besides, if the theme is compatible, what does he need to change? It should just work.

**Comment by Dh42** [comment]
And since we do not have the theme or access to the shop, we are not the correct people to reach out to.

**Comment by doclucas** [comment]
So let's just leave it broken. Wonderful approach.

**Comment by Traumflug** [comment]
We have now 13 comment and not a single screenshot demonstrating the issue.

Here's a nice tool for making and annotating screenshots: https://github.com/lupoDharkael/flameshot It's also in the Ubuntu package repository, so installation is as simple as a `sudo apt-get install flameshot`.

**Comment by Traumflug** [comment]
P.S.: one can copy such screenshots into the clipboard and paste them into a comment here. No need to deal with files.

**Comment by Dh42** [comment]
@Traumflug  Here is a link demonstrating the issue, http://no317.com/thirtybees/  I can see the product name just fine. This is a dev install of 1.0.6 and using the latest panda theme.

There is no issue and as a matter of how we develop, we do not scout other forums looking for bugs in 3rd party software or themes.

**Comment by Dh42** [comment]
Because you do not know, let me give you a bit of background. The reason why Panda is supported is because one of our community members @alwayspaws worked with Johnny on getting support added. There is simply not enough time in the day for one of the 4 developers here to test every possible theme, then make bug reports to the theme maker, who then in turn gets paid to sell the theme. 

We have reached out to him many times about support and gotten a few answers back. But as you can see in his forum, he is not even actively supporting his product by answering questions from users that have purchased his themes. This is not something we can change. This issue could also be a more than meets the eye issue as well. The product name is retrieved from the cart object at this point. With the user having the issue we have no clue if they have an override on the cart object. So the issue could actually be an incompatibility between a module and thirty bees, or even a module and the theme.

**Comment by doclucas** [comment]
Thanks for the background story, Lesley.
>The reason why Panda is supported is because one of our community members @alwayspaws worked with Johnny on getting support added. 
By "support added" you mean - make it compatible with TB, right?

**Comment by Dh42** [comment]
Correct. They had bought it with PrestaShop and worked with him letting him know of any bugs with it and he fixed them. I have tried to email him several times in the last couple of months, but he is no longer replying.

**Comment by doclucas** [comment]
Yeah I noticed he is MIA in his forums. My point tho, is not specifically about Panda, but that's just a good example - if  Thirtybees is supposed to be fully compatible with legacy (PS 1.6) themes out of the box, then why would anyone need to get support _added_? All themes should just work if they work on PS 1.6.

Get what I mean?

**Comment by Dh42** [comment]
They should work out of the box, but that does not mean that all of these themes work out of the box even with PrestaShop. There are so many variables that things are complicated. We do not remove or change pieces of the core that will break things, that is the best we can do.

**Comment by doclucas** [comment]
>They should work out of the box, but that does not mean that all of these themes work out of the box even with PrestaShop.
Obviously, but that's not what I asked. Let me rephrase - if a theme fully does work out of the box with Prestashop 1.6 (as Panda in this example did) and it is claimed the TB is fully compatible with PS 1.6 themes, then why would anyone need to get support _added_?

> We do not remove or change pieces of the core that will break things, that is the best we can do.
Never suggested or even thought that anybody here would intentionally do that, but can you guarantee such thing is impossible to have unintentionally happened?

**Comment by Traumflug** [comment]
Despite the list above has no checkmark in the Panda 2 column for thirty bees compatibility, the theme selling page states compatibility: https://www.sunnytoo.com/product/panda-creative-responsive-prestashop-theme Even with a thirty bees logo in the product picture.

**Comment by doclucas** [comment]
Although it's not really relevant to the discussion much and doesn't answer my 2 important questions (to Lesley) above, I'll respect you and answer your comment, Markus:
- The link above does have a checkmark for Panda. It doesn't have a checkmark for Transformer.
- That page clearly says `Thirtybees 1.0.3`

![image](https://user-images.githubusercontent.com/5753365/45217753-71155d00-b2cf-11e8-8628-c9a27da80e54.png)
